cranelift/filetests/filetests/isa/aarch64/bitops-run.clif

```diff
@@ -0,0 +1,266 @@
+    v5 = iconcat v2, v3
+
+    v6 = bxor v4, v5
+
+    v7, v8 = isplit v6
+    return v7, v8
+}
+; run: %bxor_i128(0, 0, 0, 0) == [0, 0]
+; run: %bxor_i128(-1, -1, 0, 0) == [-1, -1]
+; run: %bxor_i128(-1, -1, -1, -1) == [0, 0]
+; run: %bxor_i128(-1, -1, 0, -1) == [-1, 0]
+
+; run: %bxor_i128(0x01234567_89ABCDEF, 0xFEDCBA98_76543210, 0xFEDCBA98_76543210, 0x01234567_89ABCDEF) == [-1, -1]
+; run: %bxor_i128(0x8FA50A64_8FA50A64, 0x9440A07D_9440A07D, 0x4F5AE48A_4F5AE48A, 0x4A8A5F82_4A8A5F82) == [0xC0FFEEEE_C0FFEEEE, 0xDECAFFFF_DECAFFFF]
+
+
+function %band_not_i128(i64, i64, i64, i64) -> i64, i64 {
+block0(v0: i64,v1: i64,v2: i64,v3: i64):
+    v4 = iconcat v0, v1
+    v5 = iconcat v2, v3
+
+    v6 = band_not v4, v5
+
+    v7, v8 = isplit v6
+    return v7, v8
+}
+; run: %band_not_i128(0, 0, 0, 0) == [0, 0]
+; run: %band_not_i128(-1, -1, 0, 0) == [-1, -1]
+; run: %band_not_i128(-1, -1, -1, -1) == [0, 0]
+; run: %band_not_i128(-1, -1, 0, -1) == [-1, 0]
+
+; run: %band_not_i128(0x01234567_89ABCDEF, 0xFEDCBA98_76543210, 0xFEDCBA98_76543210, 0x01234567_89ABCDEF) == [0x01234567_89ABCDEF, 0xFEDCBA98_76543210]
+; run: %band_not_i128(0xF1FFFEFE_F1FFFEFE, 0xFEEEFFFF_FEEEFFFF, 0x31001010_31001010, 0x20240000_20240000) == [0xC0FFEEEE_C0FFEEEE, 0xDECAFFFF_DECAFFFF]
+
+
+function %bor_not_i128(i64, i64, i64, i64) -> i64, i64 {
+block0(v0: i64,v1: i64,v2: i64,v3: i64):
+    v4 = iconcat v0, v1
+    v5 = iconcat v2, v3
+
+    v6 = bor_not v4, v5
+
+    v7, v8 = isplit v6
+    return v7, v8
+}
+; run: %bor_not_i128(0, 0, 0, 0) == [-1, -1]
+; run: %bor_not_i128(-1, -1, 0, 0) == [-1, -1]
+; run: %bor_not_i128(-1, -1, -1, -1) == [-1, -1]
+; run: %bor_not_i128(-1, 0, 0, -1) == [-1, 0]
+
+; run: %bor_not_i128(0x01234567_89ABCDEF, 0xFEDCBA98_76543210, 0xFEDCBA98_76543210, 0x01234567_89ABCDEF) == [0x01234567_89ABCDEF, 0xFEDCBA98_76543210]
+; run: %bor_not_i128(0x80AAAAAA_80AAAAAA, 0x8A8AAAAA_8A8AAAAA, 0xBFAABBBB_BFAABBBB, 0xABBFAAAA_ABBFAAAA) == [0xC0FFEEEE_C0FFEEEE, 0xDECAFFFF_DECAFFFF]
+
+
+function %bxor_not_i128(i64, i64, i64, i64) -> i64, i64 {
+block0(v0: i64,v1: i64,v2: i64,v3: i64):
+    v4 = iconcat v0, v1
+    v5 = iconcat v2, v3
+
+    v6 = bxor_not v4, v5
+
+    v7, v8 = isplit v6
+    return v7, v8
+}
+; run: %bxor_not_i128(0, 0, 0, 0) == [-1, -1]
+; run: %bxor_not_i128(-1, -1, 0, 0) == [0, 0]
+; run: %bxor_not_i128(-1, -1, -1, -1) == [-1, -1]
+; run: %bxor_not_i128(-1, -1, 0, -1) == [0, -1]
+
+; run: %bxor_not_i128(0x01234567_89ABCDEF, 0xFEDCBA98_76543210, 0xFEDCBA98_76543210, 0x01234567_89ABCDEF) == [0, 0]
+; run: %bxor_not_i128(0x8FA50A64_8FA50A64, 0x9440A07D_9440A07D, 0xB0A51B75_B0A51B75, 0xB575A07D_B575A07D) == [0xC0FFEEEE_C0FFEEEE, 0xDECAFFFF_DECAFFFF]
+
+function %ishl_i128_i8(i64, i64, i8) -> i64, i64 {
+block0(v0: i64, v1: i64, v2: i8):
+    v3 = iconcat v0, v1
+
+    v4 = ishl.i128 v3, v2
+
+    v5, v6 = isplit v4
+    return v5, v6
+}
+; run: %ishl_i128_i8(0x01010101_01010101, 0x01010101_01010101, 2) == [0x04040404_04040404, 0x04040404_04040404]
```

Comment:
@cfallin I pretty much copied the testing code from [x64/shift-i128-run.clif](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/filetests/filetests/isa/x64/shift-i128-run.clif). And added a few more test cases where my implementation had issues.

Do you think it would be a good idea to merge run tests so that they could be multi arch? I've been thinking about something along the lines of:

`testfiles/runtests/i128-bitops.clif`:
```
test run
target x86_64
target aarch64
target s390x

function %rest_of_the_tests() {}
```

This way we could reuse all of these test cases for all arches.

The down side of this is that we may have to be more granular with the testfiles. For example this file right here fails for x86_64 because some of the bit ops fail. So we would have to split this into shift_run and bitops_run or something along those lines.

What do you think?

Reply:
Yes, sharing run-tests makes a lot of sense! In general I actually want to try to shift tests from golden-code (compile tests) to golden-output (run tests) as this makes our suite more robust against cross-cutting backend changes, such as regalloc optimizations; so there are additional benefits, aside from sharing across architectures.

Could you be more specific about the failures you're seeing on x86_64, though? That's somewhat concerning -- unimplemented opcode, incorrect result, or something else?

Reply:
Sure, band_not is failling on this [assert](https://github.com/bytecodealliance/wasmtime/blob/7d36fd9a1e070e2d83f8b63e44faae587a138686/cranelift/codegen/src/isa/x64/lower.rs#L1888).

BorNot / BxorNot / Cls (not on this PR yet) are not implemented at all.

Reply:
Ah, OK, for tests like those for `BandNot` where we don't have an implementation on x86_64, we can just omit the `target` line in the test file, perhaps with a comment saying "not yet implemented on $PLATFORM".